[Logs] New page for Logpush to HTTPS destination. (#4230)
cloudflare/cloudflare-docs: content/logs/get-started/http-destination.md

---
pcx-content-type: concept
title: Enable HTTP destination 
weight: 50
---

# Enable HTTP destination

> Cloudflare Logpush now supports the ability to send logs to arbitrary HTTP endpoints.

Reviewer: I don't think "arbitrary" is exactly the right word for this.  I know what you mean but arbitrary could also be read as random.  How about "configurable"?

> Note that when using Logpush to HTTP endpoints, Cloudflare customers are expected to perform their own authentication of the pushed logs. For example, customers may specify a secret token in the URL or an HTTP header of the Logpush destination.

Reviewer 2: It's not a requirement we enforce this from our side. I think 'may' actually fits better in this case.

## Manage via API

To create a job, make a `POST` request to the [Logpush job creation endpoint URL](/logs/reference/logpush-api-configuration/) with the appropriate parameters.

The supported parameters are as follows:

- Fields that are unchanged from other sources:
    - **dataset** (required): For example, `http_requests`.
    - **name** (optional): We suggest using your domain name as the job name.
    - **logpull_options** (optional): Refer to [Logpush API configuration options](/logs/reference/logpush-api-configuration/#options) to configure fields, sample rate, and timestamp format.
- Unique fields: 
    - **destination_conf**: Where to send the logs. This consists of an endpoint url and HTTP headers used.

>         - Any `"header_*"` URL parameters will be taken to set request headers.

Reviewer 3: ```suggestion
        - Any `"header_*"` URL parameters will be used to set request headers.
```

>             - Thus, the HTTPS endpoint cannot have custom URL parameters that conflicts with this.

Reviewer 3: Remove 'Thus' as this is a new paragraph and make 'this' into a proper noun (I am guessing that it is the URL parameters?)

```suggestion
            - The HTTPS endpoint cannot have custom URL parameters that conflicts with any `"header_*"` URL parameters you have set.
```

            - These parameters must be properly URL-encoded (that is, use `"%20"` for a whitespace), otherwise some special characters may be decoded incorrectly.
        - destination_conf may have more URL parameters in addition to special `"header_*"` parameters.

>             - Non URL-encoded special chars will be encoded when uploading.

Reviewer 3: Characters?

```suggestion
            - Non URL-encoded special characters will be encoded when uploading.
```

        - Example: `https://logs.example.com?header_Authorization=Basic%20REDACTED&tags=host:theburritobot.com,dataset:http_requests`

>     - **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB Note that there is not a way to set a minimum file size; this means that log files may be much smaller than this batch size.

Reviewer 3: ```suggestion
    - **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB. Note that there is no way to set a minimum file size; this means that log files may be much smaller than this batch size.
```

>     - **max_upload_records** (optional): The maximum number of log lines per batch. This must be at least 1000 lines or more. Note that there is not a way to specify a minimum number of log lines per batch; this means that log files may contain many fewer lines than this.

Reviewer 3: ```suggestion
    - **max_upload_records** (optional): The maximum number of log lines per batch. This must be at least 1000 lines or more. Note that there is no way to specify a minimum number of log lines per batch; this means that log files may contain many fewer lines than this.
```

{{<Aside type="note" header="Note">}}

> **Ownership_challenge** is not required.

Reviewer 3: Should this be in backticks (`)? Also kind of random, should there be more context here?
